app/components/provider_interface/provider_user_details_component.rb

```diff
@@ -26,7 +26,7 @@ def permission_rows
         {
           key: "Permissions: #{permission.provider.name}",
           value: render(PermissionsList.new(permission)),
-          change_path: provider_interface_provider_user_edit_providers_path(@provider_user),
+          change_path: provider_interface_provider_user_edit_permissions_path(@provider_user, provider_id: permission.provider.id),
           action: 'Change',
```

Comment:
Future PR: to disambiguate the links for specs & screenreaders:

```suggestion
          action: "Change permissions for #{permission.provider.name}"
```

Reply:
Decided to add this, didn't break anything.